[Jalenleeruixian] iP

Jalenleeruixian opened the pull request:
### BuddyBot

>  "Enjoy my Chatbot!" - me

Some features: 😊
- java coded
- ~slow~ fast!
- _wants you to use!_

How to view?
1. Go to **Jalenleeruixian** Github [account](https://github.com/nus-cs2103-AY2425S1/ip/pull/271). 
2. Find the ip repo and look through the data file

Features
- [x] Add tasks
- [x] Delete tasks
- [ ] Reminders

Here is the `main` method
```
public static void main(String[] args) {
        new BuddyBot("./data/BuddyBot.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jalenleeruixian committed:
Level 0

Jalenleeruixian committed:
level 0

Jalenleeruixian committed:
Level 1

Jalenleeruixian committed:
Level 1

Jalenleeruixian committed:
Level 2

Jalenleeruixian committed:
Level 3

Jalenleeruixian committed:
Level 4

Jalenleeruixian committed:
level 4

Jalenleeruixian committed:
Level 4

Jalenleeruixian committed:
Level 5

Jalenleeruixian committed:
Level 6

Jalenleeruixian committed:
Level 6

Jalenleeruixian committed:
Enums

Jalenleeruixian committed:
Added files

Jalenleeruixian committed:
test

Jalenleeruixian committed:
no message

Jalenleeruixian committed:
Level-8

Jalenleeruixian committed:
testing branch

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into branch-level-8

* master:
  test

# Conflicts:
#	src/main/java/BuddyBot.java

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'branch-level-8'

* branch-level-8:
  test
  testing branch

# Conflicts:
#	src/main/java/BuddyBot.java

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into branch-level-7

* master:
  test

# Conflicts:
#	src/main/java/BuddyBot.java

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'branch-level-7'

* branch-level-7:
  test
  test

# Conflicts:
#	src/main/java/BuddyBot.java

Jalenleeruixian committed:
Level 8

Jalenleeruixian committed:
Level 7

Jalenleeruixian committed:
More OOP

Jalenleeruixian committed:
Packages

Jalenleeruixian committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Jalenleeruixian committed:
Use Gradle

Jalenleeruixian committed:
A-JUnit

Jalenleeruixian committed:
A-Jar

Jalenleeruixian committed:
Add Java Doc

Jalenleeruixian committed:
Update to Java Doc

Jalenleeruixian committed:
JavaDoc branch

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into JavaDoc

* master:
  test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'JavaDoc'

* JavaDoc:
  test
  JavaDoc branch

# Conflicts:
#	src/main/java/buddybot/NewBuddyBot.java

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into branch-A-JavaDoc

* master:
  test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  test
  test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Coding standard

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into branch-A-CodingStandard

* master:
  test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  test
  Coding standard

Jalenleeruixian committed:
Coding standard

Jalenleeruixian committed:
Update name

Jalenleeruixian committed:
Update

Jalenleeruixian committed:
Level 9

Jalenleeruixian committed:
9

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'master' into branch-Level-9

* master:
  test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
test

Jalenleeruixian committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  test
  9

Jalenleeruixian committed:
Level 9

Jalenleeruixian committed:
Update to standard

Jalenleeruixian committed:
Update to standard

Jalenleeruixian committed:
Update to standard

Jalenleeruixian committed:
Add basic GUI

Jalenleeruixian committed:
Add Gui

Jalenleeruixian committed:
Made test

Jalenleeruixian committed:
Made test

Jalenleeruixian committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Made test
  Add Gui

Jalenleeruixian committed:
Update to GUI

Jalenleeruixian committed:
Made test

Jalenleeruixian committed:
Made test

Jalenleeruixian committed:
Merge branch 'branch-testing'

* branch-testing:
  Made test

Jalenleeruixian committed:
Add GUI

apollo-tan commented on src/main/java/buddybot/FileStorage.java:
```diff
@@ -0,0 +1,93 @@
+package buddybot;
+
+import java.io.*;
```
Considering replacing the wildcard import with the actual classes that you need to import to add clarity!

meerkatboy replied:
Agreed! That will help to reduce the possibility of potential name conflicts, such as if the same class name is found in more than one package imported through wildcard

apollo-tan commented on src/main/java/buddybot/FileStorage.java:
```diff
@@ -0,0 +1,93 @@
+           bw.write(myTasks);
+           //bw.newLine();
+           bw.close();
+           fw.close();
```
Perhaps the names fw and bw could be written in full for greater clarity!

apollo-tan commented on src/main/java/buddybot/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    /**
+     * Ass a Task to TaskLst
```
Typo!

Jalenleeruixian committed:
Test key

meerkatboy commented on src/main/java/buddybot/Ui.java:
```diff
@@ -0,0 +1,132 @@
+    /**
+     * Print a message when the end date is before the start date
+     */
+    public String showInvalidDateRange() {
```
methods are all well named with clear indications on purpose

meerkatboy commented on src/main/java/buddybot/Main.java:
```diff
@@ -0,0 +1,31 @@
+    public void start(Stage stage) {
+        try {
+            FXMLLoader myFxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
+            AnchorPane ap = myFxmlLoader.load();
```
can consider writing out ap as anchorpane in small caps, for better readability!

Jalenleeruixian committed:
Add assert

Jalenleeruixian committed:
Do test

Jalenleeruixian committed:
Do test

Jalenleeruixian committed:
Merge branch 'branch-A-Assertations'

* branch-A-Assertations:
  Do test

Jalenleeruixian committed:
Do test assert

Jalenleeruixian committed:
Make branch

Jalenleeruixian committed:
Merge pull request #1 from Jalenleeruixian/branch-A-CodeQuality

Make branch

Jalenleeruixian committed:
Change the name of variables
This will make for better readability. Currently some of the names are ambiguous.
Use words rather than short forms will improve code quality

Jalenleeruixian committed:
Improve Java docs
There are some errors presently. Amending them will improve readability.
This improves code quality.

Jalenleeruixian committed:
Add Jar file

Jalenleeruixian committed:
Add extension

Jalenleeruixian committed:
Give reference

Jalenleeruixian committed:
Add Readme

Jalenleeruixian committed:
Try screenshot

Jalenleeruixian committed:
Add screenshot

Jalenleeruixian committed:
Add screenshot

Jalenleeruixian committed:
Add user guide

Jalenleeruixian committed:
Add release
